tests/__helpers__/tasksRepositoryMock.ts

```diff
@@ -0,0 +1,28 @@
+import * as taskRepository from '../../src/repositories/tasks';
+import { functionConfig, generateFunction } from './functions';
+
+export const tasksRepositoryMock = (update = false) => {
+  const existingFunction = generateFunction(functionConfig);
+  const findData = update ? [existingFunction] : [];
+
+  const findFunctionsSpy = jest.spyOn(taskRepository.functions, 'find')
+    .mockImplementation(() => Promise.resolve(findData));
+
+  const findFunctionByNameSpy = jest.spyOn(taskRepository.functions, 'findByName')
+    .mockImplementation(() => Promise.resolve(existingFunction));
+
+  const createFunctionSpy = jest.spyOn(taskRepository.functions, 'create')
+    .mockImplementation(() => Promise.resolve(existingFunction));
+
+  const updateFunctionSpy = jest.spyOn(taskRepository.functions, 'update')
+    .mockImplementation(() => Promise.resolve(existingFunction));
```

Comment:
Could it be that either this or the code is wrong? 🤔 The code seems to imply the `update` function returns a `affectedRecords` response

Reply:
This is wrong... 😓